TypeCobol/AntlrGrammar/CobolCodeElements.g4

```diff
@@ -3843,13 +3843,13 @@ sentenceEnd:
 // Thus 2:41 PM is expressed as 14410000.
 
 acceptStatement: 
-	acceptDataTransfer | acceptSystemDateTime;
+	ACCEPT alphanumericStorageArea? (fromEnvironment? | fromSystemDateTime);
 
-acceptDataTransfer:
-	ACCEPT alphanumericStorageArea (FROM mnemonicForEnvironmentNameReferenceOrEnvironmentName)?;
+fromEnvironment:
+	FROM mnemonicForEnvironmentNameReferenceOrEnvironmentName;
 
-acceptSystemDateTime:
-	ACCEPT alphanumericStorageArea FROM ((DATE yyyyMmDd?) | (DAY yyyyDdd?) | DAY_OF_WEEK | TIME);
+fromSystemDateTime:
+	FROM ((DATE yyyyMmDd?) | (DAY yyyyDdd?) | DAY_OF_WEEK | TIME);
```

Comment:
Is it possible to have the message coming from ANTLR?
As you said: maybe by keeping the initial grammar but changing the builder only.

Reply:
Using the original grammar with a modified builder does not seem to work. We end up with unwanted calls to `EnterAcceptStatement`, themselves resulting into `AcceptStatement` code elements with no consumed tokens. Although I don't have the explanation for those extra calls, it seems that the modified grammar is more reliable, producing the correct number of code elements.

I've changed it using your suggestion on the syntax for the optional fragement.

Reply:
If the last solution works, it should be fine as it stays understandable.